src/frontend/src/modals/IOModal/components/chatView/chatInput/newChatInput.tsx

```diff
@@ -0,0 +1,214 @@
+import { usePostUploadFile } from "@/controllers/API/queries/files/use-post-upload-file";
+import useFileSizeValidator from "@/shared/hooks/use-file-size-validator";
+import useAlertStore from "@/stores/alertStore";
+import { useEffect, useRef, useState } from "react";
+import ShortUniqueId from "short-unique-id";
+import {
+  ALLOWED_IMAGE_INPUT_EXTENSIONS,
+  CHAT_INPUT_PLACEHOLDER,
+  CHAT_INPUT_PLACEHOLDER_SEND,
+  FS_ERROR_TEXT,
+  SN_ERROR_TEXT,
+} from "../../../../../constants/constants";
+import useFlowsManagerStore from "../../../../../stores/flowsManagerStore";
+import {
+  ChatInputType,
+  FilePreviewType,
+} from "../../../../../types/components";
+import FilePreview from "../filePreviewChat/newFilePreview";
+import ButtonSendWrapper from "./components/buttonSendWrapper/newButtonSendWrapper";
+import TextAreaWrapper from "./components/textAreaWrapper/newTextAreaWrapper";
+import UploadFileButton from "./components/uploadFileButton/newUploadFileButton";
+import { getClassNamesFilePreview } from "./helpers/get-class-file-preview";
+import useAutoResizeTextArea from "./hooks/use-auto-resize-text-area";
+import useFocusOnUnlock from "./hooks/use-focus-unlock";
+export default function ChatInput({
+  lockChat,
+  chatValue,
+  sendMessage,
+  setChatValue,
+  inputRef,
+  noInput,
+  files,
+  setFiles,
+  isDragging,
+}: ChatInputType): JSX.Element {
+  const currentFlowId = useFlowsManagerStore((state) => state.currentFlowId);
+  const [inputFocus, setInputFocus] = useState<boolean>(false);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const setErrorData = useAlertStore((state) => state.setErrorData);
+  const { validateFileSize } = useFileSizeValidator(setErrorData);
+
+  useFocusOnUnlock(lockChat, inputRef);
+  useAutoResizeTextArea(chatValue, inputRef);
+
+  const { mutate } = usePostUploadFile();
+
+  const handleFileChange = async (
```

Comment:
**just a question!**
can we create a separate custom hook file to handle this, just to clean up the code in the component?